tests/integration/src/test/java/org/mule/test/integration/exceptions/ErrorHandlerTestCase.java

```diff
@@ -231,6 +231,9 @@ private void callAndThrowException(Object payload, final Exception exceptionToTh
     public Event process(Event event) throws MuleException {
       Throwable exception = (Throwable) event.getVariable("exception").getValue();
       if (exception instanceof MuleException) {
+        if (exception instanceof MessagingException) {
```

Comment:
In this test case, it's creating messaging exception manually in test method, passing it as a variable and then launching it from a custom component. When it is launched it has the wrong event in it.